yt/utilities/fortran_utils.py

```diff
@@ -219,15 +219,16 @@ def skip(f, n=1, endian='='):
     >>> f = open("fort.3", "rb")
     >>> skip(f, 3)
     """
-    skipped = []
+    skipped = np.zeros(n, dtype=np.int32)
+    fmt = endian+"I"
+    fmt_size = struct.calcsize(fmt)
     for i in range(n):
-        fmt = endian+"I"
-        size = f.read(struct.calcsize(fmt))
-        s1= struct.unpack(fmt, size)[0]
-        f.seek(s1+ struct.calcsize(fmt), os.SEEK_CUR)
-        s2= struct.unpack(fmt, size)[0]
-        assert s1==s2 
-        skipped.append(s1/struct.calcsize(fmt))
+        size = f.read(fmt_size)
+        s1 = struct.unpack(fmt, size)[0]
+        f.seek(s1 + fmt_size, os.SEEK_CUR)
+        s2 = struct.unpack(fmt, size)[0]
+        assert s1 == s2
```

Comment:
I realize you didn't write this code but as a cleanup can you convert this assert into an if statement that conditionally raises an IOError with an informative error message?